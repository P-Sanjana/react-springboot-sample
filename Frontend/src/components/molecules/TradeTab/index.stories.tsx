import React from "react";
import { Story, Meta } from "@storybook/react";
import TradeTab, { TradeTabProps } from "./index";

export default {
  title: "Molecules/TradeTab",
  component: TradeTab,
} as Meta;

const Template: Story<TradeTabProps> = (args) => <TradeTab {...args} />;

export const Trade = Template.bind({});
Trade.args = {
  name: "Bitcoin",
  price: "3,285,553.73",
  change: "+1.06%",
  marketCap: "60.1T",
  watch: true,
  shortName: "BTC",
};
