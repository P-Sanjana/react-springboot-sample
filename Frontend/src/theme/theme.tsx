import { Color } from "@material-ui/core";
import { createTheme } from "@material-ui/core/styles";
import GraphikMedium from "../assets/fonts/GraphikMedium.otf";
import GraphikRegular from "../assets/fonts/GraphikRegular.otf";
import GraphikSemiBold from "../assets/fonts/GraphikSemibold.otf";

type ColorPartial = Partial<Color>;

declare module "@material-ui/core/styles/createPalette" {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  interface PaletteColor extends ColorPartial {}
}

declare module "@material-ui/core/styles/createTypography" {
  interface Typography {
    inhherit: React.CSSProperties;
    caption2: React.CSSProperties;
  }

  interface TypographyOptions {
    inherit?: React.CSSProperties;
    caption2?: React.CSSProperties;
  }
}

declare module "@material-ui/core/Typography/Typography" {
  interface TypographyPropsVariantOverrides {
    caption2: true;
  }
}

const fontFaceGraphikRegular = {
  fontFamily: "GraphikRegular",
  src: `url(${GraphikRegular}) format("opentype")`,
};

const fontFaceGraphikSemiBold = {
  fontFamily: "GraphikSemiBold",
  src: `url(${GraphikSemiBold}) format("opentype")`,
};

const fontFaceGraphikMedium = {
  fontFamily: "GraphikMedium",
  src: `url(${GraphikMedium}) format("opentype")`,
};

const theme = createTheme({
  spacing: [0, 4, 8, 12, 16, 20, 24, 32],
  palette: {
    primary: {
      main: "#0052FF",
      "100": "#FAFCFF",
      "300": "#CCE3FF",
      "500": "#0052FF",
      "700": "#002EB7",
      "900": "#00177A",
    },
    success: {
      main: "#20B03F",
      "500": "#20B03F",
      "100": "#E9F7EC",
    },
    warning: {
      main: "#FFA74F",
      "500": "#FFA74F",
      "100": "#FFF6ED",
    },
    error: {
      main: "#B71A33",
      "500": "#B71A33",
      "100": "#F3E6EB",
    },
    grey: {
      "50": "#F2F2F7",
      "100": "#E8E8F7",
      "300": "#B4B4CF",
      "500": "#4B4B60",
      "700": "#252545",
      "900": "#0E0E2E",
    },
    text: {
      primary: "#343446",
      secondary: "#7D7D89",
      hint: "#7D7D89",
      disabled: "#B2B2B9",
    },
    secondary: {
      main: "#FFA74F",
      contrastText: "#FFFFFF",
    },
  },
  overrides: {
    MuiCssBaseline: {
      "@global": {
        "@font-face": [
          fontFaceGraphikRegular,
          fontFaceGraphikSemiBold,
          fontFaceGraphikMedium,
        ],
      },
    },
  },
  typography: {
    overline: {
      gammaOverlineFontFamily: "Comic Sans Ms ",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "12px",
      lineHeight: "14px",
      identicalToBoxHeightOr117LetterSpacing: "0.005em",
      betaTextHighEmphasisColor: "#343446",
    },
    h4: {
      fontFamily: "GraphikMedium",
      fontSize: "40px",
      fontWeight: 500,
      letterSpacing: "-1%",
      lineHeight: "54px",
    },
    h6: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: 400,
      fontSize: "24px",
      lineHeight: "34px",
    },
    caption: {
      fontFamily: "GraphikMedium",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "14px",
      lineHeight: "16px",
      letterSpacing: "0.01em",
    },
    inherit: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "14px",
      lineHeight: "16px",
      letterSpacing: "0.01em",
    },
    button: {
      fontFamily: "GraphikSemiBold",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "14px",
      lineHeight: "42px",
      letterSpacing: "0.01em",
    },
    body2: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "16px",
      lineHeight: "22px",
      letterSpacing: "0.01em",
    },
    subtitle2: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "20px",
      lineHeight: "28px",
      letterSpacing: "0.005em",
    },
    subtitle1: {
      fontFamily: "GraphikSemiBold",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "20px",
      lineHeight: "28px",
      letterSpacing: "0.005em",
    },
    body1: {
      fontFamily: "GraphikSemiBold",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "16px",
      lineHeight: "22px",
      letterSpacing: "0.01em",
    },
    caption2: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: 400,
      fontSize: "14px",
      lineHeight: "16px",
      letterSpacing: "0.01em",
    },
  },
});

export default theme;
