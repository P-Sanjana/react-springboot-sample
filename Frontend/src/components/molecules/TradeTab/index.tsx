import React from "react";
import {
  makeStyles,
  Typography,
  Grid,
  Card,
  Avatar,
  Box,
} from "@material-ui/core";
import { Star, StarBorderOutlined } from "@material-ui/icons";
import Bitcoin from "../../../assets/coinImage/bitcoin.png";
import theme from "../../../theme/theme";

export type TradeTabProps = {
  name: string;
  price: string;
  change: string;
  marketCap: string;
  watch: boolean;
  shortName: string;
};

const TradeTab: React.FC<TradeTabProps> = ({
  name,
  price,
  change,
  marketCap,
  watch,
  shortName,
}) => {
  const useStyles = makeStyles({
    card: {
      display: "flex",
      flexDirection: "row",
      alignItems: "center",
      padding: "16px24px",
      height: "74px",
      border: `1px solid ${theme.palette.grey[100]}`,
      boxSizing: "border-box",
      borderRadius: "4px",
      margin: "12px0px",
      paddingTop: "20px",
    },
    nameStyle: {
      display: "flex",
      flexDirection: "row",
    },
    shortNameStyle: {
      height: "14px",
      paddingBottom: "20px",
      order: 1,
      margin: "0px 15px",
      top: "26px",
    },
    avatar: {
      width: "32px",
      height: "32px",
      left: "10px",
      borderRadius: "100px",
    },
    nameTextStyle: {
      width: "138px",
      height: "22px",
      margin: "0px 15px",
      overflow: "hidden",
    },
    priceStyle: {
      width: "170px",
      height: "22px",
      margin: "4px 0px",
      overflow: "hidden",
    },
    changeStyle: {
      width: "150px",
      height: "22px",
      left: "661px",
      top: "30px",
      color:
        parseInt(change) > 0
          ? theme.palette.success["500"]
          : theme.palette.error["500"],
      overflow: "hidden",
    },
    marketCapStyle: {
      width: "130px",
      height: "22px",
      left: "899px",
      top: "30px",
      color: theme.palette.text.primary,
      overflow: "hidden",
    },
    columnContent: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
    },
  });
  const classes = useStyles();
  return (
    <Card className={classes.card}>
      <Grid container spacing={2} xs={12}>
        <Grid item xs={3} sm={3}>
          <Box className={classes.nameStyle}>
            <Avatar className={classes.avatar} src={Bitcoin} />
            <Box className={classes.columnContent}>
              <Typography
                className={classes.nameTextStyle}
                variant="body1"
                data-testid="coinType"
              >
                {name}
              </Typography>

              <Typography
                variant="overline"
                className={classes.shortNameStyle}
                data-testid="coinTypeShort"
              >
                {shortName}
              </Typography>
            </Box>
          </Box>
        </Grid>
        <Grid item xs={3} sm={3}>
          <Typography
            className={classes.priceStyle}
            variant="body2"
            data-testid="price"
          >
            ${price}
          </Typography>
        </Grid>
        <Grid item xs={2} sm={3}>
          <Typography
            className={classes.changeStyle}
            variant="body2"
            data-testid="change"
          >
            {change}
          </Typography>
        </Grid>
        <Grid item xs={2} sm={2}>
          <Typography
            className={classes.marketCapStyle}
            variant="body2"
            data-testid="marketCap"
          >
            ${marketCap}
          </Typography>
        </Grid>
        <Grid item xs={1} sm={1}>
          {watch ? (
            <Star data-testid="watch" color="primary" />
          ) : (
            <StarBorderOutlined data-testid="watch" />
          )}
        </Grid>
      </Grid>
    </Card>
  );
};

export default TradeTab;
