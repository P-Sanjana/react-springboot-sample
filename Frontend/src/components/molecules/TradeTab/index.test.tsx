import React from "react";
import { render, screen } from "@testing-library/react";
import theme from "../../../theme/theme";
import TradeTab from "./index";

describe("TradeTab Component", () => {
  it("should have semantic2 color when change is positive", () => {
    render(
      <TradeTab
        name="Bitcoin"
        price="3,285,553.73"
        change="+1.06%"
        marketCap="60.1T"
        watch={false}
        shortName="BTC"
      />,
    );
    const comp = screen.getByTestId("change");
    expect(comp).toHaveStyle(`color:${theme.palette.success["500"]}`);
  });
  it("should have semantic6 color when change is negative", () => {
    render(
      <TradeTab
        name="Bitcoin"
        price="3,285,553.73"
        change="-1.06%"
        marketCap="60.1T"
        watch={false}
        shortName="BTC"
      />,
    );
    const comp = screen.getByTestId("change");
    expect(comp).toHaveStyle(`color:${theme.palette.error["500"]}`);
  });
  it("should match snapshot", () => {
    const wrapper = render(
      <TradeTab
        name="Bitcoin"
        price="3,285,553.73"
        change="-1.06%"
        marketCap="60.1T"
        watch={false}
        shortName="BTC"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
