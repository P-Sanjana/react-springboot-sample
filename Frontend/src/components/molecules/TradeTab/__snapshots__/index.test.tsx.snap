// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradeTab Component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-23 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3"
          >
            <div
              class="MuiBox-root MuiBox-root-32 makeStyles-nameStyle-24"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular makeStyles-avatar-26"
              >
                <img
                  class="MuiAvatar-img"
                  src="bitcoin.png"
                />
              </div>
              <div
                class="MuiBox-root MuiBox-root-33 makeStyles-columnContent-31"
              >
                <p
                  class="MuiTypography-root makeStyles-nameTextStyle-27 MuiTypography-body1"
                  data-testid="coinType"
                >
                  Bitcoin
                </p>
                <span
                  class="MuiTypography-root makeStyles-shortNameStyle-25 MuiTypography-overline"
                  data-testid="coinTypeShort"
                >
                  BTC
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3"
          >
            <p
              class="MuiTypography-root makeStyles-priceStyle-28 MuiTypography-body2"
              data-testid="price"
            >
              $
              3,285,553.73
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
          >
            <p
              class="MuiTypography-root makeStyles-changeStyle-29 MuiTypography-body2"
              data-testid="change"
            >
              -1.06%
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-2"
          >
            <p
              class="MuiTypography-root makeStyles-marketCapStyle-30 MuiTypography-body2"
              data-testid="marketCap"
            >
              $
              60.1T
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              data-testid="watch"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-23 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3"
        >
          <div
            class="MuiBox-root MuiBox-root-32 makeStyles-nameStyle-24"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular makeStyles-avatar-26"
            >
              <img
                class="MuiAvatar-img"
                src="bitcoin.png"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-33 makeStyles-columnContent-31"
            >
              <p
                class="MuiTypography-root makeStyles-nameTextStyle-27 MuiTypography-body1"
                data-testid="coinType"
              >
                Bitcoin
              </p>
              <span
                class="MuiTypography-root makeStyles-shortNameStyle-25 MuiTypography-overline"
                data-testid="coinTypeShort"
              >
                BTC
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-3"
        >
          <p
            class="MuiTypography-root makeStyles-priceStyle-28 MuiTypography-body2"
            data-testid="price"
          >
            $
            3,285,553.73
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-3"
        >
          <p
            class="MuiTypography-root makeStyles-changeStyle-29 MuiTypography-body2"
            data-testid="change"
          >
            -1.06%
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-2"
        >
          <p
            class="MuiTypography-root makeStyles-marketCapStyle-30 MuiTypography-body2"
            data-testid="marketCap"
          >
            $
            60.1T
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            data-testid="watch"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
